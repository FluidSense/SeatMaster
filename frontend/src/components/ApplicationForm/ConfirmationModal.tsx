import KnappBase from 'nav-frontend-knapper';
import Modal from 'nav-frontend-modal';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { _ALERT_USER_SUBMISSION, _SUBMIT_BUTTON_MODAL } from './strings';

interface IProps {
  modalIsOpen: boolean;
  changeModal: (modalOpen: boolean) => void;
  accept: () => void;
}

export const ConfirmationModal: React.FunctionComponent<IProps> = (props) => {
  const { changeModal, accept } = props;
  const closeModal = () => changeModal(false);
  return (
    <Modal
      isOpen={props.modalIsOpen}
      onRequestClose={closeModal}
      closeButton={false}
      contentLabel="Form submitted"
    >
      <div style={{ padding: '2rem 2.5rem' }} className="modalContent">
        {_ALERT_USER_SUBMISSION}
        <br />
        <Link to="/">
          <KnappBase type="hoved" htmlType="submit" onClick={accept} autoDisableVedSpinner={true}>
            {_SUBMIT_BUTTON_MODAL}
          </KnappBase>
        </Link>
      </div>
    </Modal>
  );
};

export default ConfirmationModal;
