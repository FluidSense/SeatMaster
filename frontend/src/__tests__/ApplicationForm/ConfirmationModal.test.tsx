import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ConfirmationModal from '../../components/ApplicationForm/ConfirmationModal';

describe('application form confirmation modal', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ConfirmationModal modalIsOpen={true} changeModal={jest.fn()}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
