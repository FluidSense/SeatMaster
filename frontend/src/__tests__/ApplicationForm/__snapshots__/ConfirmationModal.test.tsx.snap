// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application form confirmation modal renders correctly 1`] = `
<ModalWrapper
  closeButton={false}
  closeTimeoutMS={0}
  contentLabel="Form submitted"
  isOpen={true}
  onRequestClose={[Function]}
  shouldCloseOnOverlayClick={true}
>
  <div
    className="modalContent"
    style={
      Object {
        "padding": "2rem 2.5rem",
      }
    }
  >
    Your application has been submitted and will be processed. You can see its status on the home page.
    <br />
    <Link
      replace={false}
      to="/"
    >
      <KnappBase
        autoDisableVedSpinner={true}
        className={null}
        disabled={false}
        htmlType="submit"
        inaktivKlasseVedDisabled={false}
        mini={false}
        spinner={false}
        type="hoved"
      >
        Go home
      </KnappBase>
    </Link>
  </div>
</ModalWrapper>
`;
